src/wallet/mod.rs

```diff
@@ -836,23 +838,40 @@ where
             current_height,
         )?;
 
-        // get drain script
-        let drain_script = match params.drain_to {
-            Some(ref drain_recipient) => drain_recipient.clone(),
-            None => self
-                .get_internal_address(AddressIndex::New)?
-                .address
-                .script_pubkey(),
+        // prepare the drain script
+        let weighted_drain_script = {
+            let script_pubkey = match params.drain_to {
+                Some(ref drain_recipient) => drain_recipient.clone(),
+                None => self
+                    .get_internal_address(AddressIndex::New)?
+                    .address
+                    .script_pubkey(),
+            };
+
+            let satisfaction_weight = if params.drain_to.is_none() {
+                self.get_descriptor_for_keychain(KeychainKind::Internal)
+                    .max_satisfaction_weight()
+                    .unwrap()
+            } else {
+                0
```

Comment:
Didn't we consider the `drain_to` script pubkey as something external? the satisfaction weight for the wallet is zero because it won't spend the created output.

Reply:
> Didn't we consider the drain_to script pubkey as something external?

Yeah

> the satisfaction weight for the wallet is zero because it won't spend the created output.

I admit I'm a bit undecided on this... But I think we should still consider the satisfaction weight here. Even tho it's not us spending the output, someone still will have to, so I think it's fair to try to reduce their waste.

Reply:
Ok, I've to find a way to compute the `satisfaction_weight` for an arbitrary `Script`, I've tried before without success.

Reply:
How is this going? Do you need help?

Reply:
I'm not advancing in figuring out how to solve this. I've tried to implement a solution using miniscript functions without success. Another idea, which I don't like, is to pass the foreign weight as a parameter of the transaction.
Also, I don't want to implement something to parse the script and compute the weight. It would be pointless because miniscript implements those functions. But there isn't an implementation for an arbitrary scriptpubkey.